src/Microsoft.OData.Core/Json/JsonReader.cs

```diff
@@ -553,6 +553,23 @@ public virtual async Task<object> GetValueAsync()
             return this.nodeValue;
         }
 
+        /// <summary>
+        /// Asynchronously checks whether or not the current value can be streamed.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous operation.
+        /// true if the current value can be streamed; otherwise false.</returns>
+        public Task<bool> CanStreamAsync()

```

Comment:
Why does this method need to be async if doesn't do any async work?

Reply:
@habbes In `JsonReader` it just returns `true` or `false` but in the `BufferingJsonReader` it actually does call an asynchronous method - `GetValueAsync`. I had actually not considered having an asynchronous equivalent of `CanStream` method initially but I realized I couldn't avoid it when implementing asynchronous support in `BufferingJsonReader`